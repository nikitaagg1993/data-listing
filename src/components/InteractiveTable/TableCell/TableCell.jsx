import React from 'react';
import PropTypes from 'prop-types';
import TableCellStyle from './styled';
import { getProperty } from '../../../util';

const TableCell = ({
  id,
  column,
  row,
}) => (
  <TableCellStyle
    id={id}
    column={column}
  >
    {getProperty(row, column.id)}
  </TableCellStyle>
);

TableCell.propTypes = {
  id: PropTypes.string.isRequired,
  column: PropTypes.shape({
    id: PropTypes.string,
  }).isRequired,
  row: PropTypes.shape({}).isRequired,
};

export default TableCell;
