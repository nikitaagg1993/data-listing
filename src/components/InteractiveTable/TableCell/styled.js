import styled, { css } from 'styled-components';
// import { Cell } from './Cell';
import { media } from '../../shared/media';

const overflowCSS = css`
  div:first-child {
    white-space: ${(props) => (props.column.wrap ? 'normal' : 'nowrap')};
    overflow: ${(props) => (props.column.allowOverflow ? 'visible' : 'hidden')};
    text-overflow: ellipsis;
  }
`;

export const CellBase = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  ${({ theme, head }) => theme && theme[head ? 'headCells' : 'cells'] && theme[head ? 'headCells' : 'cells'].style};
  ${(props) => props.noPadding && 'padding: 0'};
`;

// Flex calculations
export const Cell = styled(CellBase)`
 

  ${(props) => props.column.width && css`
     width: ${props.column.width};
  `};

  ${(props) => !props.column.width && css`
    flex-grow: 1;
    flex-shrink: 0;
    flex-basis: 0;
  `};

  ${(props) => props.column.numeric && css`
    justify-content: flex-end;
  `};
  ${(props) => props.column.right && 'justify-content: flex-end'};
  ${(props) => (props.column.center || props.column.button) && 'justify-content: center'};
  ${(props) => (props.column.compact || props.column.button) && 'padding: 0'};

  /* handle hiding cells */
  ${(props) => props.column.hide && props.column.hide === 'sm' && media.sm`
    display: none;
  `};
  ${(props) => props.column.hide && props.column.hide === 'md' && media.md`
    display: none;
  `};
  ${(props) => props.column.hide && props.column.hide === 'lg' && media.lg`
    display: none;
  `};
  ${(props) => props.column.hide && Number.isInteger(props.column.hide) && media.custom(props.column.hide)`
    display: none;
  `};
`;

const TableCellStyle = styled(Cell)`
  font-size: ${(props) => props.theme.rows && props.theme.rows.fontSize};
  font-weight: 400;
  ${(props) => !props.column.cell && overflowCSS};
  ${(props) => props.column.style};
  ${(props) => props.extendedCellStyle};
`;

export default TableCellStyle;
