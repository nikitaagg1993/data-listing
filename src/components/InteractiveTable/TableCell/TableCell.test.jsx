import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import TableCell from './TableCell';

test('Table Cell', () => {
  const column = { id: 'name' };
  const row = { name: 'title' };

  const tree = renderer.create(<TableCell
    id="id"
    column={column}
    row={row}
  />);
  const testInstance = tree.root;

  expect(tree.toJSON()).toMatchSnapshot();
  expect(testInstance.findAllByType('div').length).toEqual(1);
  expect(testInstance.props).toMatchObject({ id: 'id', column: { id: 'name' }, row: { name: 'title' } });
});
