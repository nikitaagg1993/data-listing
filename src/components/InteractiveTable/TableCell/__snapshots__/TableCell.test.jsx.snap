// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Cell 1`] = `
.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
}

.c1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
}

.c2 {
  font-weight: 400;
}

.c2 div:first-child {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<div
  className="c0 c1 c2"
  id="id"
>
  title
</div>
`;
